### ISSUE TITLE
Enforce line breaks in between types and variables, functions, or classes

### ISSUE THREAD

**Issue by mcmire** [MetaMask/eslint-config#335]
These should produce violations:

``` typescript
type X = "string";
const foo = "bar";
```

``` typescript
type X = "string";
const foo = () => {
  // do whatever
}
```

``` typescript
type X = "string";
function foo() {
  // do whatever
}
```

``` typescript
type X = "string";
class A {
 // ...
}
```
